### PR TITLE
Add columntable method for OptSummary

### PR DESCRIPTION
**Pull request by dmbates**

### CODE REVIEW

File: src/optsummary.jl [hunk 1]
```diff
@@ -90,6 +90,34 @@ function OptSummary(
     )
 end
 
+"""
+    columntable(s::OptSummary, [stack::Bool=false])
+
+Return `s.fitlog` as a `Tables.columntable`.
+
+When `stack` is false (the default), there will be 3 columns in the result:
+- `iter`: the iteration number
```
Comment (palday) [inline src/optsummary.jl:99]: It's technically the sample number, i.e. `iteration / thin`

File: test/pls.jl [hunk 2]
```diff
@@ -132,7 +132,21 @@ end
     # since we're caching the fits, we should get it back to being correctly fitted
     # we also take this opportunity to test fitlog
     @testset "fitlog" begin
+        thin = 1
+        refit!(fm1; REML=false, progress=false, thin)
         fitlog = fm1.optsum.fitlog
+        fitlogtbl = columntable(fm1.optsum)
+        @test length(fitlogtbl) == 3
+        @test keys(fitlogtbl) == (:iter, :objective, :θ)
+        @test length(first(fitlogtbl)) > 15   # can't be sure of exact length
```
Comment (palday) [inline test/pls.jl:141]: since `thin=1`, this should be just `feval`

File: src/optsummary.jl (hunk 1 above)
Comment (dmbates) [reply src/optsummary.jl]: Changed

**Review by palday — APPROVED** [review]

File: test/pls.jl (hunk 2 above)
Comment (dmbates) [reply test/pls.jl:141]: We'll defer on this one a bit because presently it is `feval + 1`

**Pull request merged**